src/main/java/ru/test/restservice/api/FileApiController.java

```diff
@@ -47,4 +47,9 @@ public void saveFiles(@RequestBody List<FileItemDTO> files) throws IOException {
     public void deleteFile(@RequestParam String path) throws IOException {
         fileService.deleteFile(path);
     }
+
+    @PutMapping("/new-file")
```

Comment:
```suggestion
    @PostMapping("/projects/{projectId}/files/")
```
И это объединить с перезаписью
По идее тебе вообще разницы никакой не будет -- создать или перезаписать